server/routes/popVerifyRoutes.ts

```diff
@@ -1,41 +1,72 @@
 import { type RequestHandler, Router } from 'express'
 import path from 'path'
 import asyncMiddleware from '../middleware/asyncMiddleware'
-import setUpMultipartFormDataParsing from '../middleware/setUpMultipartFormDataParsing'
 
 export default function routes(): Router {
   const router = Router()
   const get = (routePath: string | string[], handler: RequestHandler) => router.get(routePath, asyncMiddleware(handler))
   const post = (routePath: string | string[], handler: RequestHandler) =>
     router.post(routePath, asyncMiddleware(handler))
-  router.use(setUpMultipartFormDataParsing())
 
   get('/', async (req, res, next) => {
     res.render('pages/pop-id/submit-photo')
   })
 
   get('/options', async (req, res, next) => {
     // re-showing previous choices on going back is a nice to have for a later iteration - use session stored values.
-    res.render('pages/pop-id/options')
+    res.render('pages/pop-id/options', { selectedOption: req.session.idOption })
   })
 
   post('/options', async (req, res, next) => {
     const { option } = req.body
     req.session.idOption = option
 
     if (option === 'upload') {
-      res.redirect(`/pop/verify/upload-photo`)
+      return res.redirect(`/pop/verify/upload-photo`)
     }
-    res.redirect(`/pop/verify/take-photo`)
+    return res.redirect(`/pop/verify/take-photo`)
   })
 
   get('/upload-photo', async (req, res, next) => {
-    res.render('pages/pop-id/upload')
+    const { errorMessage } = req.session
+    delete req.session.errorMessage
+    res.render('pages/pop-id/upload-photo', { errorMessage })
+  })
+
+  post('/upload-photo', async (req, res, next) => {
+    const { file } = req
+    if (!file) {
+      req.session.errorMessage = 'No file uploaded'
+      return res.redirect('/pop/verify/upload-photo')
+    }
+    const relativePath = `/assets/uploads/${req.session.user_id}/${path.basename(file.path)}`
+
+    req.session.verificationPhoto = relativePath
+    return res.redirect('/pop/verify/display-photo')
+  })
+
+  get('/display-photo', async (req, res, next) => {
+    const { verificationPhoto } = req.session
+    if (!verificationPhoto) {
```

Comment:
This should redirect to options so that after a "try again" the user can take a photo if there's nothing suitable to upload and vise versa